### PR TITLE
Bring `hotshot-types` and `hs-builder-api` into `Hotshot`

### PR DESCRIPTION
**Pull request by bfish713**
Merge the 3 repos

Reviewers please check if I forgot to add something(probably this would be outside of the `/src` directories) or messed up something in a Cargo.toml file

### CODE REVIEW

**Review by move47 — APPROVED** [review]
Thanks @bfish713 for this.

File: crates/builder-api/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[package]
+name = "hs-builder-api"
```
Comment (ss-es) [inline crates/builder-api/Cargo.toml:2]: For consistency I feel like we should probably call this `hotshot-builder-api` (I think everything else related to HotShot is called `hotshot-`). But I think we should do that separately, in another PR.

@move47 Thoughts?

Comment (move47) [reply crates/builder-api/Cargo.toml:2]: Yeah, I had this also. We have to change in several other crates too, so thought of doing altogether.

**Review by ss-es — APPROVED** [review]

**Review by move47 — DISMISSED** [review]
LGTM! Thanks.

File: crates/builder-api/Cargo.toml (hunk 1 above)
Comment (bfish713) [reply crates/builder-api/Cargo.toml]: Oh I added it to this pr, hopefully that's ok

**Review by jparr721 — APPROVED** [review]

**Pull request merged**